File: Scala/src/Sources/Selection/selection.scala
package Sources.Selection

import Others.swap.swap
import Others.value._

object selection {
    private def selection(a: Array[Int]): Unit = {
        for (i <- 0 until a.length - 1) {
            var m = i
            for (j <- i + 1 until a.length) {
                if (a(j) < a(m))
                    m = j
            }

            if (m != i)
                swap(a, m, i)
        }
    }

    def main(args: Array[String]): Unit = {
        val s = 1000 // Changeable size
        val a = rev_p(s)

        val start = System.currentTimeMillis()

        selection(a)

        val end = System.currentTimeMillis() - start

        for (i <- a.indices)
            println(a(i))
        print(end + " ms")
    }
}

File: Kotlin/src/main/kotlin/Sources/Selection/selection.kt
fun selection(a: IntArray) {
    for (i in a.indices-1) {
        var m = i
        for (j in i+1 until a.size) {
            if (a[j] < a[m])
                m = j
        }

        if (m != i)
            swap(a, m, i)
    }
}

fun main() {
    val a = intArrayOf(5, 4, 3, 2, 1)

    val start = System.currentTimeMillis()

    selection(a)

    val end = System.currentTimeMillis() - start

    for (i in a.indices)
        println(a[i])
    print("$end ms")
}